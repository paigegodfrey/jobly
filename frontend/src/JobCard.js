import React from 'react';
import './Card.css';

const JobCard = ({ job = {}, handleApply }) => {
  const { title, salary, equity } = job;

  let formatCurrency = new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
  });

  let formattedSalary = formatCurrency.format(salary).slice(0, -3);
  let formattedEquity = Math.round(equity * 100, 0) + '%';

  return (
    <div className="col">
      <div className="JobCard Card card">
        <div className="card-body">
          <h6 className="card-title d-flex justify-content-between">
            <span className="text-capitalize">{title}</span>
          </h6>
          <div>Salary: {formattedSalary}</div>
          <div>Equity: {formattedEquity}</div>
          <button
            className="job-btn btn btn-outline-secondary font-weight-bold"
            onClick={handleApply}
            disabled={job.state}
          >
            {job.state ? "Applied" : "Apply"}
          </button>
        </div>
      </div>
    </div>
  );
}

export default JobCard;