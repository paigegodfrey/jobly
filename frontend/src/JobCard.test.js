import React from "react";
import { render } from "@testing-library/react";
import JobCard from "./JobCard";

it("renders without crashing", function() {
  render(<JobCard />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<JobCard />);
  expect(asFragment()).toMatchSnapshot();
});