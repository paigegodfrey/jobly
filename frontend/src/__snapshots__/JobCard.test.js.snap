// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col"
  >
    <div
      class="JobCard Card card"
    >
      <div
        class="card-body"
      >
        <h6
          class="card-title d-flex justify-content-between"
        >
          <span
            class="text-capitalize"
          />
        </h6>
        <div>
          Salary: $
        </div>
        <div>
          Equity: NaN%
        </div>
        <button
          class="job-btn btn btn-outline-secondary font-weight-bold"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
